search-service/src/main/kotlin/com/egm/stellio/search/listener/IAMListener.kt

```diff
@@ -99,9 +105,20 @@ class IAMListener(
         tenantName: String,
         entityDeleteEvent: EntityDeleteEvent
     ): Either<APIException, Unit> = either {
+        val subjectType = SubjectType.valueOf(entityDeleteEvent.entityTypes.first().uppercase())
+        val sub = entityDeleteEvent.entityId.extractSub()
         mono {
             subjectReferentialService.delete(entityDeleteEvent.entityId.extractSub())
         }.writeContextAndSubscribe(tenantName, entityDeleteEvent)
+        if (searchProperties.ownerDeleteCascadeEntities && subjectType == SubjectType.USER) {
+            val entitiesIds = entityAccessRightsService.getEntitiesIdsOwnedBySubject(
+                sub
+            )
+                .getOrNull()
+            entitiesIds?.forEach { entityId ->
+                entityPayloadService.deleteEntity(entityId)
+            }
```

Comment:
you could add a new `entityPayloadService.deleteEntities(...)` function

Reply:
Batch Entity Delete is a seperate feature (see `5.6.10` ), it would be better to implement it in a seperate PR.

Reply:
this was not my point (and btw, batch entity delete is already implemented in Stellio).

my point was that, instead of calling `entityPayloadService.deleteEntity(entityId)` for each entity, add a new function `entityPayloadService.deleteEntities(...)` that accepts a list of entities

Reply:
btw, `entityOperationService.delete()` is already doing the job